src/counter_analysis_toolkit/flops.c

```diff
@@ -329,14 +395,58 @@ void exec_double_cholesky( int EventSet, FILE *fp ) {
         /* Stop and print count. */
         resultline( n, CHOLESKY, EventSet, fp );
 
-        keep_double_cholesky_res( n, ld );
+        keep_double_mat_res( n, ld );
     }
 
     free( ad );
     free( ld );
 }
 
-void keep_double_norm_res( int n, double *xd ) {
+void exec_double_gemm( int EventSet, FILE *fp ) {
+
+    int i, j, n, retval;
+    double *ad=NULL, *bd=NULL, *cd=NULL;
+
+    /* Print info about the computational kernel. */
+    print_header( fp, "Double-Precision", "GEMM" );
+
+    /* Allocate the matrices. */
+    ad = malloc( MAXDIM * MAXDIM * sizeof(double) );
+    bd = malloc( MAXDIM * MAXDIM * sizeof(double) );
+    cd = malloc( MAXDIM * MAXDIM * sizeof(double) );
+
+    /* Step through the different array sizes. */
+    for ( n = 0; n < MAXDIM; n++ ) {
+        /* Initialize the needed arrays at this size. */
+        for ( i = 0; i < n; i++ ) {
+            for ( j = 0; j < n; j++ ) {
+                cd[i * n + j] = 0.0;
+                ad[i * n + j] = ((double)rand())/((double)RAND_MAX) * (double)1.1;
```

Comment:
Can you use "random()" instead of "rand()". It's supposed to be the better function.